packages/core/strapi/lib/commands/__tests__/data-transfer/transfer.test.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+const utils = require('../../transfer/utils');
+
+const mockDataTransfer = {
+  createRemoteStrapiDestinationProvider: jest.fn(),
+  createLocalStrapiSourceProvider: jest.fn(),
+  createTransferEngine: jest.fn().mockReturnValue({
+    transfer: jest.fn().mockReturnValue(Promise.resolve({})),
+  }),
+};
+
+jest.mock(
+  '@strapi/data-transfer',
+  () => {
+    return mockDataTransfer;
+  },
+  { virtual: true }
+);
+
+const transferCommand = require('../../transfer/transfer');
+
+const exit = jest.spyOn(process, 'exit').mockImplementation(() => {});
+const logger = jest.spyOn(console, 'error').mockImplementation(() => {});
+
+jest.mock('../../transfer/utils');
+
+const destinationUrl = 'ws://strapi.com';
+
+describe('transfer', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('uses destination url provided by user without authentication', async () => {
+    await transferCommand({ from: 'local', to: destinationUrl });
+
+    expect(mockDataTransfer.createRemoteStrapiDestinationProvider).toHaveBeenCalledWith(
+      expect.objectContaining({
+        url: destinationUrl,
+      })
+    );
+
+    expect(exit).toHaveBeenCalled();
```

Comment:
I just discovered a few days ago we're mocking exit wrong, because it's actually quite tricky to get right (since normally it would halt execution, but when mocked it continues the flow). Take a look at the latest version of export.test.js `expectExit` to see what I've come up with, that halts the flow and lets you check the exit code.